### ISSUE TITLE
Using version 5.0.1 getting "MalformedInput - Start of list found where not expected"

### ISSUE THREAD

**Issue by justindantzer** [keithbrink/amazon-mws-laravel#44]
Admittedly first time using MWS, and also this package.

Note: This request does work in the MWS Scratchpad

Trying out the simple example to get orders with:

```php
$amz = new \KeithBrink\AmazonMws\AmazonOrderList('store1');
$amz->setLimits('Modified', "13:00", "16:00");
$amz->setFulfillmentChannelFilter("MFN");
$amz->setOrderStatusFilter([
    "Unshipped",
    "PartiallyShipped",
    "Canceled",
    "Unfulfillable",
]);
$amz->setUseToken();
$amz->fetchOrders();

dd($amz->getList());
```

I modified `AmazonOrderList.php` to output some of the request info in Laravel:
```
https://mws.amazonservices.com/Orders/2013-09-01
    ?SellerId=A16W_____________
    &MarketplaceId=ATVP_____________
    &AWSAccessKeyId=AKIA_____________
    &MWSAuthToken=amzn.mws._____________
    &SignatureVersion=2
    &SignatureMethod=HmacSHA256
    &Version=2013-09-01
    &MarketplaceId.Id.1=ATVP_____________
    &LastUpdatedAfter=2020-08-04T12%3A58%3A00%2B00%3A00
    &LastUpdatedBefore=2020-08-04T15%3A58%3A00%2B00%3A00
    &FulfillmentChannel.Channel.1=MFN
    &OrderStatus.Status.1=Unshipped
    &OrderStatus.Status.2=PartiallyShipped
    &OrderStatus.Status.3=Canceled
    &OrderStatus.Status.4=Unfulfillable
    &Action=ListOrders
    &Timestamp=2020-08-04T12%3A14%3A39%2B00%3A00
    &Signature=%2BlV9NeZS3AXM5JTTtyDeBfTUcYwNMtQRp4ztylWM2oE%3D
```

Compared to the scratchpad request:
```
/Orders/2013-09-01
    ?AWSAccessKeyId=AKIA_____________
      &Action=ListOrders
      &SellerId=A16W_____________
      &MWSAuthToken=amzn.mws._____________
      &SignatureVersion=2
      &Timestamp=2020-08-03T20%3A59%3A25Z
      &Version=2013-09-01
      &Signature=xZVEFV%2BjWGdWmeD8mWJ5UR7EmcG0ixNPxEI%2FJhQ9ktg%3D
      &SignatureMethod=HmacSHA256
      &LastUpdatedAfter=2020-08-03T17%3A00%3A00Z
      &LastUpdatedBefore=2020-08-03T20%3A00%3A00Z
      &MarketplaceId.Id.1=ATVP_____________
      &OrderStatus.Status.1=Unshipped
      &OrderStatus.Status.2=PartiallyShipped
      &OrderStatus.Status.3=Canceled
      &OrderStatus.Status.4=Unfulfillable
      &FulfillmentChannel.Channel.1=MFN HTTP/1.1
```

The response I'm getting in Laravel:
```xml
<?xml version="1.0"?>
<ErrorResponse xmlns="https://mws.amazonservices.com/Orders/2013-09-01">
  <Error>
    <Type>Sender</Type>
    <Code>MalformedInput</Code>
    <Message>Start of list found where not expected</Message>
  </Error>
  <RequestId>3dded5bf-44da-496a-9ea3-b59a3e2f8802</RequestId>
</ErrorResponse>
```

Looking at the requests, I note that the request from the package has two references to MarketplaceId, referencing both `MarketplaceId` & `MarketplaceId.Id.1`, whereas the Scratchpad request only references `MarketplaceId.Id.1`.

I tried removing the reference to `MarketplaceId`, but then I receive a new error: 
`InvalidParameterValue - The input you have submitted is not valid. Please check your input and try again.`

This leads me to believe that isn't the problem, but is the only difference that stands out.

Any help would be greatly appreciated.

**Comment by justindantzer** [comment]
[This previous issue](https://github.com/keithbrink/amazon-mws-laravel/issues/20) seems relevant, but implementing this change results in the noted `InvalidParameterValue` error.

**Comment by justindantzer** [comment]
> [This previous issue](https://github.com/keithbrink/amazon-mws-laravel/issues/20) seems relevant, but implementing this change results in the noted `InvalidParameterValue` error.

After extracting code to simply run the curl command, I was able to create a working example:
```php
$uri = 'https://mws.amazonservices.com/Orders/2013-09-01';

$fields = [
    "AWSAccessKeyId" => "AKIA_____________",
    "Action" => "ListOrders",
    "SellerId" => "A16W_____________",
    "MWSAuthToken" => "amzn.mws._____________",
    "SignatureVersion" => 2,
    "Version" => "2013-09-01",
    "SignatureMethod" => "HmacSHA256",
    "LastUpdatedAfter" => "2020-06-03T17:00:00Z",
    "LastUpdatedBefore" => "2020-08-03T20:00:00Z",
    "MarketplaceId.Id.1" => "ATVP_____________",
    "OrderStatus.Status.1" => "Unshipped",
    "OrderStatus.Status.2" => "PartiallyShipped",
    "OrderStatus.Status.3" => "Canceled",
    "OrderStatus.Status.4" => "Unfulfillable",
    "FulfillmentChannel.Channel.1" => "MFN",
    "Timestamp" => date('c', time() - 120),
];

uksort($fields, 'strcmp');

$query = http_build_query($fields, '', '&', PHP_QUERY_RFC3986);

$endpoint = parse_url($uri);

$toSign = "POST\n{$endpoint['host']}\n{$endpoint['path']}\n{$query}";

$signature = urlencode(base64_encode(hash_hmac('sha256', $toSign, 'SECRET', true)));

$query .= '&Signature=' . $signature;

$ch = curl_init();
curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
curl_setopt($ch, CURLOPT_TIMEOUT, 0);
curl_setopt($ch, CURLOPT_FORBID_REUSE, 1);
curl_setopt($ch, CURLOPT_FRESH_CONNECT, 1);
curl_setopt($ch, CURLOPT_HEADER, 1);
curl_setopt($ch, CURLOPT_URL, $uri);
curl_setopt($ch, CURLOPT_POSTFIELDS, $query);

$data = curl_exec($ch);
curl_close($ch);

dd($data);
```

This led me to modifying `AmazonOrderList`, per one of the other issue fixes, adding this to the constructor:
`unset($this->options['MarketplaceId']);`

Then in `AmazonCore.php` I moved the `uksort` out of the `_calculateStringToSignV2()` method, and into the `genQuery()` method, like:
```php
protected function genQuery()
{
    // ...

    unset($this->options['Signature']);
    $this->options['Timestamp'] = $this->genTime();

    // this line
    uksort($parameters, 'strcmp');

    $this->options['Signature'] = $this->_signParameters($this->options, $secretKey);

    return $this->_getParametersAsString($this->options);
}
```

Submitted PR #45

**Comment by keithbrink** [comment]
Resolved by #45
